### PR TITLE
Closes #454 - Support multiple applications using a single Fluo instance

### LINKED ISSUE #454: Support multiple applications using a single Fluo instance
Currently, each Fluo instance only supports one application.  It would be nicer if it could support multiple instances.  Configuration would be be structured as follows:
```bash
# Example configuration
conf/example
# Default config
conf/
# App configuration
apps/<appName>/conf
# Application observer jars
apps/<appName>/observers
```
The fluo command will have the following options:
```bash
# Initialize 'hello' application
fluo init hello
# Starts 'hello' application
fluo start hello
fluo stop hello
fluo scan hello
# If 'hello' is only application, no need to specify
fluo start
fluo stop
fluo scan
# List all available applications
fluo list   
# Print status of 'hello' application
fluo status hello
# Print status of all applications
fluo status
```

The `minifluo` command will have the following options:
``` bash
# Start hello app in MiniFluo
minifluo start hello 
# Stop
minifluo stop hello
```

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java [hunk 1]
```diff
@@ -192,17 +195,28 @@ public void validate() {
     getWorkerMaxMemory();
     getWorkerNumCores();
     getWorkerThreads();
-    getZookeepers();
     getZookeeperTimeout();
   }
 
-  public FluoConfiguration setZookeepers(String zookeepers) {
+  public FluoConfiguration setApplicationName(String applicationName) {
```
Comment (keith-turner) [inline modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java]: should probably check the app name does not have any chars that are not allowed in a zookeeper node name

File: modules/distribution/src/main/scripts/fluo [hunk 2]
```diff
@@ -27,34 +27,98 @@ script=$( basename "$SOURCE" )
...
+    exit 1
+  fi
+}
+
+function print_usage {
+  echo -e "Usage: fluo <command> (<argument> ...)\n"
+  echo -e "Possible commands:\n"
+  echo "  new <app>     Creates configuraiton for new application in apps/"
+  echo "  init <app>    Initializes a Fluo application using conf/fluo.properties"
+  echo "  start <app>   Starts Fluo application on cluster"
```
Comment (keith-turner) [inline modules/distribution/src/main/scripts/fluo]: info command not in help

File: modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java (hunk 1 above)
Comment (mikewalch) [inline modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java]: sounds good... will do.

**Comment by keith-turner** [comment]
I saw a few problem when trying to run stress w/ local commands.

The error message when there is no app given could be improved

```
$ ./bin/local-fluo start-worker 
ERROR - Fluo application  needs a directory in apps/ with a conf/ and lib/ directory
```

The worker could not find stress classes.

```
17:37:39.382 [pool-7-thread-7] ERROR i.f.c.worker.NotificationProcessor - Failed to process work 08:h8kq:08:0000000017e9f4d0 count wait 
java.lang.RuntimeException: java.lang.ClassNotFoundException: io.fluo.stress.trie.NodeObserver
	at io.fluo.core.worker.Observers.getObserver(Observers.java:78) ~[fluo-core-1.0.0-beta-1-SNAPSHOT.jar:1.0.0-beta-1-SNAPSHOT]
	at io.fluo.core.worker.WorkTask.run(WorkTask.java:51) ~[fluo-core-1.0.0-beta-1-SNAPSHOT.jar:1.0.0-beta-1-SNAPSHOT]
	at io.fluo.core.worker.NotificationProcessor$2.run(NotificationProcessor.java:125) ~[fluo-core-1.0.0-beta-1-SNAPSHOT.jar:1.0.0-beta-1-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_72]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_72]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_72]
Caused by: java.lang.ClassNotFoundException: io.fluo.stress.trie.NodeObserver
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366) ~[na:1.7.0_72]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355) ~[na:1.7.0_72]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.7.0_72]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354) ~[na:1.7.0_72]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425) ~[na:1.7.0_72]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[na:1.7.0_72]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358) ~[na:1.7.0_72]
	at java.lang.Class.forName0(Native Method) ~[na:1.7.0_72]
	at java.lang.Class.forName(Class.java:191) ~[na:1.7.0_72]
	at io.fluo.core.worker.Observers.getObserver(Observers.java:73) ~[fluo-core-1.0.0-beta-1-SNAPSHOT.jar:1.0.0-beta-1-SNAPSHOT]
	... 5 common frames omitted

```


Kill did not work.

```
$ ./bin/local-fluo kill-worker
Usage: grep [OPTION]... PATTERN [FILE]...
Try `grep --help' for more information.
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
$ ./bin/local-fluo kill-oracle
Usage: grep [OPTION]... PATTERN [FILE]...
Try `grep --help' for more information.
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]

```

**Comment by keith-turner** [comment]
Something I noticed, I don't think it has to be fixed.   Wanted to mention it in case you cared to fix it.  This is probably a script vs java issue.   The Usage below does not have app name like `Usage: fluo scan <app> [options]`
```
$ ./bin/fluo scan stress -h
Usage: fluo scan [options]
  Options:
    -h, -help, --help
       Prints help
       Default: false
    -c
       Columns of scan in comma separated format:
       <<columnfamily>[:<columnqualifier>]{,<columnfamily>[:<columnqualifier>]}>
    -e
       End row (inclusive) of scan
    -p
       Row prefix to scan
    -r
       Exact row to scan
    -s
       Start row (inclusive) of scan
```

File: docs/production-install.md [hunk 3]
```diff
@@ -111,43 +106,75 @@ Source your `.bashrc` for the changes to take effect and test the script
...
+When configuring the observer section in fluo.properties, you can configure your instance
+for the [phrasecount] application if you have not created your own application. See
+the [phrasecount] example for instructions. You can also choose not to configure any
+observers but your workers will be idle when started.
 
-Initializing Fluo
------------------
+The `apps/myapp/lib` directory should contain any observer jars for your application. If 
+you configured [fluo.properties] for observers, copy any jars containing these
+observer classes this directory.
```
Comment (keith-turner) [inline docs/production-install.md]: `copy any jars containing these observer classes this directory`

??

**Comment by keith-turner** [comment]
I did not get a chance to test mini with stress, if that works +1

File: modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java (hunk 1 above)
Comment (mikewalch) [inline modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java]: fixed in b51333acb243e

File: modules/distribution/src/main/scripts/fluo (hunk 2 above)
Comment (mikewalch) [inline modules/distribution/src/main/scripts/fluo]: added in b51333acb243e

File: docs/production-install.md (hunk 3 above)
Comment (mikewalch) [inline docs/production-install.md]: fixed in b51333acb243e

**Comment by mikewalch** [comment]
In b51333acb243e, also fixed several minor issues with `fluo` & `mini-fluo`

**Pull request merged**